### PR TITLE
Add shell.nix for Nix(OS) users

### PR DESCRIPTION
**Pull request by enderger**
This PR adds a `shell.nix` file for Nix(OS) users. In Nix, a `shell.nix` file is used to create a development environment with certain packages installed, in this case just Golang.

### CODE REVIEW

**Comment by driusan** [comment]
Thanks! Unfortunately, I've never used NixOS so I'm not really capable of reviewing this.

Is this a manually created file or one that's generated by some kind of tool? If it's manually created, maybe it could be tangled from a markdown block. That way, the text could include some explanation and links to relevant documentation included for people who are new to Nix (like myself.)

**Comment by enderger** [comment]
This is a manually created file, I’ll explain:
1. `{ pkgs ? import <nixpkgs> {} }:` tells Nix to store the package collection as a variable called `pkgs`
2. `pkgs.mkShell` uses that variable to allow you to use `nix-shell` to allow developers to access a shell with the environment defined in the brackets.
3. `buildInputs = with pkgs; [ go ];` tells Nix to have Golang installed in the shell.

**Comment by enderger** [comment]
I added a version which can be compiled from `lmt`, describing what it does.

File: Nix-Shell.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Nix Shell
+Nix development shells can be used to get development environments which can be accessed using the `nix-shell` command.
```
Comment (driusan) [inline Nix-Shell.md:2]: I think this is still lacking context to understand it. Is there Nix documentation you could link to that goes into more depth about what it means to have a nix development shell or what the benefits are?

Comment (enderger) [reply Nix-Shell.md:2]: I'll get on that soon, have had other projects to work on.

File: SubdirectoryFiles.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ a little help from the Go standard library `path/filepath` functions.
 
 We'll get the directory of the string, and if it's not "." (filepath.Dir
 returns ".", not "" for empty paths) call `os.MkdirAll` on it in order to create
-the directory before creating the file.
+the directory before creating the file
```
Comment (driusan) [inline SubdirectoryFiles.md:12]: Why did you remove this punctuation?

**Comment by driusan** [comment]
I'm still not sure I understand what the shell.nix is for, but I'm willing to take it on faith that it's useful for Nix users and merge this after the unrelated punctuation change is fixed.

You can also add yourself to the credits at the end of the README if you'd like.

File: SubdirectoryFiles.md (hunk 2 above)
Comment (enderger) [reply SubdirectoryFiles.md:12]: Ah, sorry about that. My master branch had some temporary changes made while testing the actions PR.
